### PR TITLE
fix gradient accumulation steps without using deepspeed

### PR DESCRIPTION
**Pull request by baoyf4244**
# What does this PR do?

when training without deepspeed,  before loss backward, the loss is not scaling with  gradient_accumulation_steps,  this is caused by the parameter gradient_accumulation_steps not passed to Accelerator when creating this object.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by muellerzr — CHANGES_REQUESTED** [review]
Please try using main again, https://github.com/huggingface/transformers/pull/35651 has fixed this. 

In the future, the trainer explicitly does *not* rely on the Accelerator’s highest GA interface because it doesn’t account (easily) for the token nuances needed that Unsloth found. (We use the second/mid level API instead)

**Comment by baoyf4244** [comment]
Ok, I'm sorry I have not find that issue a few days ago.
